frontend/src/Editor/Components/Icon.jsx

```diff
@@ -0,0 +1,44 @@
+import React, { useState, useEffect } from 'react';
+// eslint-disable-next-line import/no-unresolved
+import * as Icons from '@tabler/icons';
+import { resolveWidgetFieldValue } from '@/_helpers/utils';
+
+export const Icon = ({ properties, styles, fireEvent, width, height, currentState, registerAction }) => {
+  const { icon, visibility } = properties;
+  const { iconColor } = styles;
+  const IconElement = Icons[icon];
+
+  const [hideIcon, setIconVisibility] = useState(false);
+
+  useEffect(() => {
+    setIconVisibility(
+      typeof visibility !== 'boolean' ? !resolveWidgetFieldValue(visibility, currentState) : !visibility
+    );
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [visibility]);
+
+  registerAction('click', async function () {
+    fireEvent('onClick');
+  });
+
+  registerAction('hide', async function (hide) {
+    setIconVisibility(typeof hide !== 'boolean' ? resolveWidgetFieldValue(hide, currentState) : hide);
+  });
```

Comment:
Wouldn't this get triggered on every render cycle?

Reply:
Yes, it will be triggered on every render. We can move it `useEffect` with the dependencies. 
cc: @sherfin94

Reply:
The second argument of `registerAction` takes in a dependency array that acts just like how useEffect works. We might need to revise the architecture of CSA in the near future though. But for the time being you can just pass any dependencies as the second argument of `registerAction` and it will re-register whenever the dependency gets updated.